### PR TITLE
Delete unnecessary seek from FileStream.SetLengthInternal on Unix

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]
Looks good to me once CI is green.  I'm trying to remember why all this was deemed necessary and coming up empty.

**Comment by jkotas** [comment]
I suspect that it was just copy&paste from Windows where we can avoid it as well - https://github.com/dotnet/runtime/pull/43863#issuecomment-719871894.

**Comment by stephentoub** [comment]
```
    System.IO.Tests.FileStream_SetLength.InvalidLengths [FAIL]
      Assert.Throws() Failure
      Expected: typeof(System.ArgumentOutOfRangeException)
      Actual:   (No exception was thrown)
      Stack Trace:
        /_/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs(69,0): at System.AssertExtensions.Throws[T](String expectedParamName, Action action)
        /_/src/libraries/System.IO.FileSystem/tests/FileStream/SetLength.cs(19,0): at System.IO.Tests.FileStream_SetLength.InvalidLengths()
```
```
    System.IO.MemoryMappedFiles.Tests.MemoryMappedFileTests_CreateNew.TooLargeCapacity_Unix [FAIL]
      Expected one of: (System.IO.IOException, System.ArgumentException) -> Actual: (System.ArgumentOutOfRangeException)
      Stack Trace:
        /_/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs(212,0): at System.AssertExtensions.ThrowsAnyInternal(Action action, Type[] exceptionTypes)
        /_/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs(222,0): at System.AssertExtensions.ThrowsAny[IOException,ArgumentException](Action action)
        /_/src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateNew.Tests.cs(115,0): at System.IO.MemoryMappedFiles.Tests.MemoryMappedFileTests_CreateNew.TooLargeCapacity_Unix()
        /_/src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs(378,0): at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

File: src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateNew.Tests.cs [hunk 1]
```diff
@@ -112,7 +112,7 @@ public void TooLargeCapacity_Unix()
             // due to differences in OS behaviors and Unix not actually having a notion of
             // a view separate from a map.  It could also come from CreateNew, depending
             // on what backing store is being used.
-            Assert.Throws<IOException>(() =>
+            AssertExtensions.ThrowsAny<IOException, ArgumentException>(() =>
```
Comment (stephentoub) [inline src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateNew.Tests.cs:115]: It looks like this method is poorly named, in that whereas `Assert.ThrowsAny<T>` allows for the exception to be T or anything derived from it, `AssertExtensions.ThrowsAny<T1, T2>` is requiring the concrete type of T1 or T2:
https://github.com/dotnet/runtime/blob/22c3105575d6edf545c55e1773ff8b75b766f8b5/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L195-L216
which is then failing if the exception is actually an ArgumentOutOfRangeException.

**Pull request merged**